libs/design-system/src/atoms/inputs/selectInputs.tsx

```diff
@@ -50,46 +42,49 @@ export type SelectInputProps = {
   /**
    * The function that handles the change of the input.
    */
-  handleChange: React.ChangeEventHandler<HTMLSelectElement>;
+  handleChange: (value: string) => void;
 
   /**
    * The options of the input. It's an array of options to put in the dropdown select of type OptionProps.
    */
   options: SelectOptionsProps[];
 };
 
-const variations = {
-  primary: 'plasmo-text-gray plasmo-border-gray focus:plasmo-border-cyan focus:plasmo-ring-cyan plasmo-w-full',
-  secondary:
-    'plasmo-text-3.5 plasmo-text-white plasmo-bg-ocean-blue plasmo-border-ocean-blue focus:plasmo-border-green focus:plasmo-ring-green plasmo-w-2/5',
-};
+export const SelectInput = ({ id, value, className, handleChange, options }: SelectInputProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [selectedOption, setSelectedOption] = useState<string | undefined>(value);
+
+  const toggleDropdown = () => setIsOpen(!isOpen);
+
+  const handleOptionClick = (value: string) => {
+    setSelectedOption(value);
+    handleChange(value);
+    setIsOpen(false);
+  };
 
-export const SelectInput = ({
-  type = 'primary',
-  id,
-  required,
-  value,
-  className,
-  handleChange,
-  options,
-}: SelectInputProps) => {
   return (
-    <select
-      required={required}
-      id={id}
-      name={id}
-      className={cn(
-        'plasmo-antialiased plasmo-border plasmo-border-solid plasmo-rounded-md plasmo-p-2 focus:plasmo-outline-none',
-        variations[type],
-        className
+    <div className={cn('plasmo-relative', className)} id={id}>
```

Comment:
Risky to decide not to use a real inpu as if the form became more complex we won't be able to use the input API. Not a big deal if you think it's worth it to reduce complexity 👌

Reply:
I dreaded doing this but had to... If I did, it's because I found it near impossible to properly design the dropdown select.

I'd suggest moving along, and reassess if something breaks.

Okfy?

Reply:
Okpm! That's why no one never re-design the dropdown 😅 And keeps it ugly but maximise compatibility. But we're developing a product to link LinkedIn to Notion I think we're safe 😉